examples/lua/nats_adapter.lua

```diff
@@ -0,0 +1,92 @@
+                          ["eventTime"] = request.Time,
+                          ["eventName"] = eventName,
+                          ["userIdentity"] = {
+                              ["principalId"] = request.User.Id
+  						},
+                          ["requestParameters"] = {
+                              ["sourceIPAddress"] = ""
+  						},
+                          ["responseElements"] = {
+                              ["x-amz-request-id"] =  request.Id,
+                              ["x-amz-id-2"] = request.RGWId
+  						},
+                          ["s3"] = {
+                              ["s3SchemaVersion"] = "1.0",
+                              ["configurationId"] = configure,
+                              ["bucket"] = {
+                                  ["name"] = request.Bucket.Name,
+                                  ["ownerIdentity"] = {
+                                      ["principalId"] = request.Bucket.User.Id
+                                  },
+                                  ["arn"] = "arn:aws:s3:" .. request.ZoneGroup.Name .. "::" .. request.Bucket.Name,
+                                  ["id"] = request.Bucket.Id
+  						    },
+                              ["object"] = {
+                                  ["key"] = request.Object.Name,
+                                  ["size"] = request.Object.Size,
+                                  ["eTag"] = "", -- eTag is not supported yet
+                                  ["versionId"] = request.Object.Instance,
+                                  ["sequencer"] = string.format("%x", os.time()), 
+                                  ["metadata"] ={
+  				    request.HTTP.Metadata
+                                  },
+                                  ["tags"] = {
+                                      request.Tags
+                                  }
+                              }
+                          },
+                          ["eventId"] = "",
+                          ["opaqueData"] = opaqueData
+                      }
+      }
+      return notification
+  end
+  
+  supported_event = false
+  configure = "mynotif1"
+  opaqueData = "me@example.com"
+  topic = "Bucket_Notification"
+  bucket_name = "mybucket"
+  nats_host = '0.0.0.0'
+  nats_port = 4222
+  
+  --Object Created
+  if Request.RGWOp == "put_obj" then
+      notification = toJson(Request ,'ObjectCreated:Put', opaqueData, configure)
+  elseif Request.RGWOp == "post_obj" then
+      notification = toJson(Request ,'ObjectCreated:Post', opaqueData, configure)
+  
+  elseif Request.RGWOp == "copy_obj" then
+      notification = toJson(Request ,'ObjectCreated:Copy', opaqueData, configure)
+  
+  --Object Removed
+  elseif Request.RGWOp == "delete_obj" then
+      notification = toJson(Request ,'ObjectRemoved:Delete', opaqueData, configure)
+  end
+  
+  if supported_event == true and bucket_name == Request.Bucket.Name then
```

Comment:
no point in building the json messages if the bucket in not the right one
so, probably add this check around lines 72-84